crates/memory-testing/src/main.rs

```diff
@@ -0,0 +1,41 @@
+use std::{io::Read, str::FromStr};
+
+use bitwarden_crypto::{AsymmetricCryptoKey, SymmetricCryptoKey};
+
+fn main() {
+    let now = std::time::Instant::now();
+
+    let mut test_string = String::new();
+    test_string.push_str("THIS IS USED TO CHECK THAT ");
+    test_string.push_str("THE MEMORY IS DUMPED CORRECTLY");
+
+    // In HEX:
+    // KEY: 15f8 5554 ff1f 9852 1963 55a6 46cc cf99 1995 0b15 cd59 5709 7df3 eb6e 4cb0 4cfb
+    // MAC: 4136 481f 8581 93f8 3f6c 5468 b361 7acf 7dfb a3db 2a32 5aa3 3017 d885 e5a3 1085
+    let symm_key = SymmetricCryptoKey::from_str(
+        "FfhVVP8fmFIZY1WmRszPmRmVCxXNWVcJffPrbkywTPtBNkgfhYGT+D9sVGizYXrPffuj2yoyWqMwF9iF5aMQhQ==",
```

Comment:
Do you think we should search the dump for the base64 encoded string to? To ensure we clean up `from_str`. I think that means moving away from FromStr to From<String> to ensure we consume the strings. We also have to use some direct memory calls to replace the existing string memory. Probably better left for the future.

Reply:
Yeah sounds reasonable, we can add the test for it now, and we can just ignore the failing result until we have dealt with it later

Reply:
Should we add a comment so we don't get confused why it's disabled and being triggered.

Reply:
Added the comment in the analyze program